sys/net/vnet.h

```diff
@@ -288,8 +288,10 @@ extern struct sx vnet_sxlock;
     static t VNET_NAME(n) __section(VNET_SETNAME) __used
 #endif
 #ifdef __CHERI_PURE_CAPABILITY__
-#define	_VNET_PTR(b, n)		(__typeof(VNET_NAME(n))*)		\
-	((b) + ((vaddr_t)&VNET_NAME(n) - (vaddr_t)VNET_START))
+#define	_VNET_PTR(b, n)						\
+	cheri_setbounds((__typeof(VNET_NAME(n))*)((b) +		\
```

Comment:
```suggestion
	cheri_setbounds((__typeof(VNET_NAME(n)) *)((b) +	\
```
While this is pre-existing, I feel that we'd have a space if the `__typeof()` were expanded to a literal type so there should be one here.

Reply:
Does not need to be done for this PR, but we probably will want exact bounds on these at some point?

Reply:
That might be very hard to achieve.  Right now the template section is built from individual members.  We'd have to have some way that those individual member variables were all given suitable alignment and padding at the time the final section is assembled by the linker.  However, I suspect that nearly all (probably all) of these variables are sub-page in size and in practice setboundsexact would probably be fine.

Reply:
This may be one thing to keep in mind if we add linker support for making sections representable for sandboxing purposes.

Reply:
Agreed